src/linear_algebra/eigenspace.lean

```diff
@@ -341,5 +348,115 @@ begin
       ih, ←linear_map.ker_comp, linear_map.comp_assoc], }
 end
 
+/-- Generalized eigenrange and generalized eigenspace for exponent `findim K V` are disjoint. -/
+lemma generalized_eigenvec_disjoint_range_ker
+  [field K] [vector_space K V] [finite_dimensional K V] (f : End K V) (μ : K) :
+  disjoint (f.generalized_eigenrange μ (findim K V)) (f.generalized_eigenspace μ (findim K V))  :=
+begin
+  have h : submodule.comap ((f - am μ) ^ findim K V) (f.generalized_eigenspace μ (findim K V)) =
+    (f.generalized_eigenspace μ (findim K V)),
+    from calc
+      submodule.comap ((f - am μ) ^ findim K V) (f.generalized_eigenspace μ (findim K V))
+         = ((f - am μ) ^ findim K V * (f - am μ) ^ findim K V).ker :
+        by { rw [generalized_eigenspace, ←linear_map.ker_comp], refl }
+     ... = f.generalized_eigenspace μ (findim K V + findim K V) :
+        by { rw ←pow_add, refl }
+     ... = f.generalized_eigenspace μ (findim K V) :
+        by { rw generalized_eigenspace_eq_generalized_eigenspace_findim_of_le, linarith },
+  erw [disjoint, submodule.map_inf_eq_map_inf_comap, top_inf_eq, h],
```

Comment:
`erw` is a bit of an antipattern, because it suggests the existing `rw` lemmas are not enough. In this case it's not too bad (it only saves an unfolding of `generalized_eigenrange`), unless you expect that the expression `f.generalized_eigenrange μ (findim K V) ⊓ p` will occur more.